pkg/virt-handler/virtwrap/schema.go

```diff
@@ -85,6 +108,7 @@ type Domain struct {
 
 type DomainStatus struct {
 	Status LifeCycle
+	Reason LifeCycle
```

Comment:
Regarding to the lost state, you are right. The direction looks good to me.

Any reason you prefer the untyped constants here? Wouldn't it be very easy, by accident, to assign Lifecycle states to a Reason or the other way round?